src/engine/renderer/glsl_source/processSurfaces_cp.glsl

```diff
@@ -108,7 +112,7 @@ void main() {
 	                              + gl_GlobalInvocationID.x
 	                              + 1; // Add 1 because the first surface command is always reserved as a fake command
 	// Each surfaceBatch encompasses 64 surfaceCommands with the same material, padded to 64 as necessary
-	const uvec2 materialID = surfaceBatches[globalGroupID / 2].materialIDs[globalGroupID % 2];
+	const uvec2 materialID = UVEC2_ID_TO_UVEC4( surfaceBatches, globalGroupID );
```

Comment:
The name of this macro doesn't make sense to me. Maybe `UVEC2_FROM_UVEC4_ARRAY`

Reply:
Yeah, I was looking for a better name for it.

Reply:
Done.